xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/XWiki.java

```diff
@@ -1552,24 +1552,16 @@ public Date getResourceLastModificationDate(String name)
 
     public byte[] getResourceContentAsBytes(String name) throws IOException
     {
-        InputStream is = null;
         if (getEngineContext() != null) {
-
-            try {
-                is = getResourceAsStream(name);
+            try (InputStream is = getResourceAsStream(name)) {
+                if (is == null) {
+                    return FileUtils.readFileToByteArray(new File(name));
+                }
+                return IOUtils.toByteArray(is);
             } catch (Exception e) {
```

Comment:
You should not catch all exceptions like this. This method is supposed to throw is there is an issue.

Reply:
@tmortagne What do you mean?

Reply:
I mean remove this catch. The previous version was not catching anything.